src/components/StateBrowserNetwork.vue

```diff
@@ -105,61 +105,113 @@
-                        class="kiwi-statebrowser-channels-option"
-                        @click="toggleAddChannel()"
+                        v-if="network.state === 'connected'"
+                        class="kiwi-statebrowser-channels-options"
                     >
-                        <i class="fa fa-plus" aria-hidden="true" />
+                        <div
+                            class="kiwi-statebrowser-channels-header"
+                            @click="toggleSection(type)"
+                        >
+                            <div class="kiwi-statebrowser-channel-name">
+                                <i v-if="(section_display_channels === true &&
+                                       type === 'channels') ||
+                                       (section_display_queries === true && type === 'queries')"
+                                   class="fa fa-caret-down kiwi-statebrowser-channels-toggle"
+                                />
+                                <i v-else
+                                   class="fa fa-caret-right kiwi-statebrowser-channels-toggle" />
+                                {{ type === 'channels' ? $t('channels') : $t('messages') }}
+                            </div>
+                            <div
+                                v-if="type === 'channels'"
+                                :class="{ active: channel_add_display == true }"
+                                class="kiwi-statebrowser-channels-option"
+                                @click="toggleAddChannel()"
+                            >
+                                <i class="fa fa-plus" aria-hidden="true" />
+                            </div>
+                            <div
+                                v-if="type === 'channels'"
+                                :class="{ active: channel_filter_display == true }"
+                                class="kiwi-statebrowser-channels-option"
+                                @click="onSearchChannelClick"
+                            >
+                                <i v-if="type === 'channels'"
+                                   class="fa fa-search" aria-hidden="true"
+                                />
+                            </div>
+                            <div class="kiwi-statebrowser-buffer-actions">
+                                <div class="kiwi-statebrowser-channel-labels">
+                                    <div
+                                        v-if="(!section_display_channels === true &&
+                                            type === 'channels') ||
+                                            (!section_display_queries === true &&
+                                                type === 'queries')"
+                                        class="kiwi-statebrowser-channel-label"
+                                    >
+                                        {{ buffersUnread(itemBuffers) }}
```

Comment:
I'm unsure how to manage the 0 and the highlights here without adding multiple calls to buffersUnread().
@prawnsalad is it better to move buffersUnread() to the computed section ? If so, I don't see how to catch the filteredBuffers.

Reply:
Discussed this on IRC.